### DISCUSSION TITLE
RabbitMQ deployment with third-party certificate

### DISCUSSION THREAD

**Discussion by imGannicus** [rabbitmq/rabbitmq-server#12164] [Questions]
Can rabbitmq be deployed on server with wildcard third-party certificate (e.g. godaddy) having name different from server name. Server name is like rabbitserver.qa.local and wild card certificate name is like *.qa.test.rabbit.com

**Comment by lukebakken** [comment]
> Server name is like rabbitserver.qa.local and wild card certificate name is like *.qa.test.rabbit.com

No, that's not how wildcard certs work, in any case.

**Reply by imGannicus** [reply]
wildcard certificate works for websites/web services when we get DNS entry created like rabbitserver.qa.test.rabbit.com and bind to website/web service but its not working in case of rabbitmq

**Reply by lukebakken** [reply]
> its not working in case of rabbitmq

That statement tells me absolutely nothing useful.

**Reply by imGannicus** [reply]
how can we make rabbitmq work with third-party cert?

**Reply by michaelklishin** [reply]
@imGannicus all certificates are "third party" ones for RabbitMQ. We [will not troubleshoot TLS](https://github.com/rabbitmq/rabbitmq-server/blob/main/COMMUNITY_SUPPORT.md) for non-paying users.

There are two extensive guides related to TLS:

 * On how [TLS is used](https://www.rabbitmq.com/docs/ssl) with RabbitMQ
 * On how to [troubleshoot TLS-related issues](https://www.rabbitmq.com/docs/troubleshooting-ssl), namely how to narrow them down efficiently
 * An example [tool that generates self-signed certificate chains](https://github.com/rabbitmq/tls-gen/blob/main/basic/openssl.cnf#L72-L80) which is used extensively by the RabbitMQ team for integration tests

The docs even mention some [specific log messages](https://www.rabbitmq.com/docs/ssl#when-peer-verification-fails) for very common failure scenarios, and explains how TLS peer verification works.

The only thing I can add to the doc guides above: it might be that your certificate does not have the appropriate [Subject Alternative Name](https://github.com/rabbitmq/tls-gen/blob/main/basic/openssl.cnf#L72-L80) fields. You haven't provided any details, so that's just a guess with an OpenSSL config example.

We will not guess as to what you are trying and what exactly does "its not working" mean. Such problem definitions in a domain as nuanced as TLS do nothing but waste everyone's time.
